bin/beta_diversity.py

```diff
@@ -0,0 +1,192 @@
+import os
+import numpy as np
+
+# import data_readers
+# from tree import TreeWrapper
+# import phylo_beta_diversity
+
+
+from analyses.lm_objects.tree import TreeWrapper
+from analyses.helpers import data_readers
+from analyses.phylo_beta_diversity import phylo_beta_diversity
+
+DESCRIPTION = """\
+Computes phylogenetic & ecological beta diversity components for Sorensen and Jaccard Indices."""
+
+
+# .............................................................................
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description=DESCRIPTION)
+
+    parser.add_argument(
+        'in_tree_filename', type=str, help='Path to the tree file')
+    parser.add_argument(
+        'in_tree_schema', type=str, help='The format of the tree',
+        choices=['newick', 'nexml', 'nexus'])
+
+    parser.add_argument(
+        'pam_filename', type=str,
+        help='Path to file with presence/ absence data (PAM)')
+    parser.add_argument(
+        'data_format', type=str, help='The format of the PAM',
+        choices=['csv', 'json', 'phylip', 'table'])
+
+    # Outputs
+    # Annotated tree or trees
+    # Plots
+    # Matrix csv
+
+# Might use this for nodal beta diversity 
+
+    # parser.add_argument(
+    #     'out_tree_filename', type=str,
+    #     help='Path to write the resulting annotated tree')
+    # parser.add_argument(
+    #     'out_tree_schema', type=str,
+    #     help='The format to use when writing the tree',
+    #     choices=['newick', 'nexml', 'nexus'])
+    # parser.add_argument(
+    #     '-l', '--annotate_labels', type=str,
+    #     help='If provided, annotate the tree labels with this data column')
+    # parser.add_argument(
+    #     '-p', '--plot_directory', type=str,
+    #     help='If provided, write distribution plots to this directory')
+
+    parser.add_argument(
+        'family_name', type=str,
+        help='Beta diversity family metric to calculate',
+        choices=['sorensen', 'jaccard'])
+
+    parser.add_argument(
+        'out_foldername', type=str,
+        help='Write the output of beta diversity calculations to this folder')
+
+    parser.add_argument(
+        '-n', '--number_permutations', 
+        help='The number of permuatations to calculate')
+
+    parser.add_argument(
+        '-a', '--alpha',
+        help='The alpha value to determine significance') # make default = 0.05
+
+    args = parser.parse_args()
+
+    # Number of iterations
+    if args.number_permutations == None:
+        nrand = 10
+    else:
+        nrand = int(args.number_permutations)
+    print(nrand)
+
+    # Check that input files exist
+    if not os.path.exists(args.in_tree_filename):
+        raise IOError(
+            'Input tree {} does not exist'.format(args.in_tree_filename))
+    if not os.path.exists(args.pam_filename):
+        raise IOError(
+            'Input data file {} does not exist'.format(args.pam_filename))
+
+    # Read data
+    if args.data_format == 'csv':
```

Comment:
There is an easier way to read the PAM.  You can use the the Matrix.load_csv classmethod from lm_objects.matrix.

You can replace lines 106-143 with:
```
    with open(args.pam_filename) as in_file:
        pam = Matrix.load_csv(in_file, num_header_rows=1, num_header_cols=1)
```

That should cover the test data.  If other problems come up with csvs with different numbers of headers I can work those out.